src/components/tl/dpu/tl_dpu_coll.c

```diff
@@ -0,0 +1,213 @@
+    // req_param.mask          = 0/* XXX */;
+    // req_param.mem_type      = task->args.src.info.mem_type;
+    // req_param.memory_type   = ucc_memtype_to_ucs[mtype]; /* XXX */
+
+    /* IN PLACE CHECK special in place func in ucc_coll_utils */
+    put_req[0] = ucp_put_nbx(ctx->ucp_ep,
+                                task->args.src.info.buffer,
+                                task->args.src.info.count * ucc_dt_size(task->args.src.info.datatype),
+                                team->rem_data_in, team->rem_data_in_key, &req_param);
+    if (tl_dpu_ucp_req_check(put_req[0]) != UCC_OK) {
+        return UCC_ERR_NO_MESSAGE;
+    }
+    ucp_worker_fence(ctx->ucp_worker);
+
+    put_req[1] = ucp_put_nbx(ctx->ucp_ep, &task->sync, sizeof(task->sync),
+                              team->rem_ctrl_seg, team->rem_ctrl_seg_key,
+                              &req_param);
+    if (tl_dpu_ucp_req_check(put_req[1]) != UCC_OK) {
+        return UCC_ERR_NO_MESSAGE;
+    }
+
+    do {
+        ucp_worker_progress(ctx->ucp_worker);
+    } while((tl_dpu_ucp_req_test(&(put_req[0]), ctx->ucp_worker) != UCC_OK) ||
+            (tl_dpu_ucp_req_test(&(put_req[1]), ctx->ucp_worker) != UCC_OK));
+ 
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_dpu_allreduce_init(ucc_tl_dpu_task_t *task)
+{
+    if (task->args.mask & UCC_COLL_ARGS_FIELD_USERDEFINED_REDUCTIONS) {
+        tl_error(UCC_TL_TEAM_LIB(task->team),
+                 "userdefined reductions are not supported yet");
+        return UCC_ERR_NOT_SUPPORTED;
+    }
+    if (!UCC_IS_INPLACE(task->args) && (task->args.src.info.mem_type !=
+                                        task->args.dst.info.mem_type)) {
+        tl_error(UCC_TL_TEAM_LIB(task->team),
+                 "assymetric src/dst memory types are not supported yetpp");
+        return UCC_ERR_NOT_SUPPORTED;
+    }
+
+    task->super.post     = ucc_tl_dpu_allreduce_start;
+    task->super.progress = ucc_tl_dpu_allreduce_progress;
+    return UCC_OK;
+}
+
+static ucc_status_t ucc_tl_dpu_coll_finalize(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_dpu_task_t *task = ucc_derived_of(coll_task, ucc_tl_dpu_task_t);
+    tl_info(task->team->super.super.context->lib, "finalizing task %p", task);
+    ucc_tl_dpu_free_task(task);
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_dpu_coll_init(ucc_base_coll_args_t *coll_args,
+                                         ucc_base_team_t      *team,
+                                         ucc_coll_task_t     **task_h)
+{
+    ucc_tl_dpu_team_t    *tl_team = ucc_derived_of(team, ucc_tl_dpu_team_t);
+    ucc_tl_dpu_task_t    *task    = ucc_tl_dpu_alloc_task();
+    ucc_status_t          status  = UCC_OK;
+
+    fprintf(stdout, "Collective init");
+    ucc_coll_task_init(&task->super);
+    tl_info(task->team->super.super.context->lib, "task %p initialized", task);
+
+    memcpy(&task->args, &coll_args->args, sizeof(ucc_coll_args_t));
+
+    task->sync.itt              = tl_team->coll_id;
```

Comment:
maybe give the same name to both vars: either "itt" or "coll_id" - for clarity for readers

Reply:
fixed